app/models/course.rb

```diff
@@ -370,6 +381,16 @@ def next_recruitment_cycle?
     recruitment_cycle.year > RecruitmentCycle.current_recruitment_cycle.year
   end
 
+  # Ideally this would just use the validation, but:
+  # https://github.com/rails/rails/issues/13971
+  def entry_requirements_assignable(course_params)
+    course_params.slice(:maths, :english, :science)
+      .to_h
+      .select { |_subject, value| value && !ENTRY_REQUIREMENT_OPTIONS.key?(value.to_sym) }
```

Comment:
Should this be `value.valid?`? Could it be an empty string?

Reply:
It can be nil but not an empty string. Passing in an empty string returns an error as expected.